lib/ae_mdw/dex.ex

```diff
@@ -18,125 +19,99 @@ defmodule AeMdw.Dex do
+    with {:ok, cursor} <- deserialize_swaps_cursor(cursor) do
       state
-      |> build_streamer(
-        @dex_swap_tokens_table,
-        key_boundary(@dex_swap_tokens_table, create_txi),
-        cursor
-      )
-      |> Collection.paginate(pagination, & &1, &serialize_cursor/1)
-      |> then(fn paginated_swaps -> {:ok, paginated_swaps} end)
+      |> build_swaps_streamer(cursor)
+      |> Collection.paginate(pagination, &render_swap(state, &1), &serialize_cursor/1)
+      |> then(&{:ok, &1})
     end
   end
 
-  @spec validate_token(String.t()) :: {:ok, pos_integer()} | {:error, Error.t()}
-  def validate_token(token_symbol) do
-    case DexCache.get_token_pair_txi(token_symbol) do
-      nil -> {:error, ErrInput.NotAex9.exception(value: token_symbol)}
-      create_txi -> {:ok, create_txi}
-    end
-  end
-
-  @spec fetch_account_swaps(State.t(), account_query(), pagination(), cursor()) ::
-          {:ok, paginated_account_swaps()} | {:error, Error.t()}
-  defp fetch_account_swaps(state, query, pagination, cursor) do
-    with {:ok, cursor} <- deserialize_account_cursor(cursor) do
-      state
-      |> build_streamer(@account_swaps_table, key_boundary(query), cursor)
-      |> Collection.paginate(pagination, & &1, &serialize_cursor/1)
-      |> then(fn paginated_swaps -> {:ok, paginated_swaps} end)
-    end
-  end
+  defp build_account_swaps_streamer(state, account_pk, %{create_txi: create_txi}, cursor) do
+    key_boundary = {
+      {account_pk, create_txi, Util.min_int(), nil},
+      {account_pk, create_txi, Util.max_int(), nil}
+    }
 
-  @spec fetch_contract_swaps(State.t(), txi(), pagination(), cursor()) ::
-          {:ok, paginated_contract_swaps()} | {:error, Error.t()}
-  defp fetch_contract_swaps(state, create_txi, pagination, cursor) do
-    with {:ok, cursor} <- deserialize_contract_cursor(cursor) do
-      state
-      |> build_streamer(@contract_swaps_table, key_boundary(create_txi), cursor)
-      |> Collection.paginate(pagination, & &1, &serialize_cursor/1)
-      |> then(fn paginated_swaps -> {:ok, paginated_swaps} end)
-    end
-  end
+    cursor =
+      case cursor do
+        nil -> nil
+        {_account_pk, _create_txi, txi, log_idx} -> {account_pk, create_txi, txi, log_idx}
+      end
 
-  defp build_streamer(state, @account_swaps_table = table, boundary, cursor) do
     fn direction ->
-      Collection.stream(state, table, direction, boundary, cursor)
+      Collection.stream(state, @account_swaps_table, direction, key_boundary, cursor)
     end
   end
 
-  defp build_streamer(state, @contract_swaps_table = table, boundary, cursor) do
+  defp build_account_swaps_streamer(state, account_pk, _query, cursor) do
+    key_boundary = {
+      {account_pk, Util.min_int(), nil, nil},
+      {account_pk, Util.max_int(), nil, nil}
+    }
+
+    cursor =
+      case cursor do
+        nil -> nil
+        {_account_pk, create_txi, txi, log_idx} -> {account_pk, create_txi, txi, log_idx}
```

Comment:
It would be weird if the first account_pk is not the account_pk given, but that would only happen if the user is altering the cursor, so I guess it wouldn't matter. Removed it